openwisp_firmware_upgrader/admin.py

```diff
@@ -14,7 +14,6 @@
 from django.templatetags.static import static
 from django.urls import resolve, reverse
 from django.utils.html import format_html
-from django.utils.safestring import mark_safe
```

Comment:
Why are you removing this?

Reply:
./openwisp_firmware_upgrader/admin.py:17:1: F401 'django.utils.safestring.mark_safe' imported but unused

I changed mark_safe to format_html 
as mark_safe can lead to XSS vulnerabilities if used with untrusted input

Reply:
The[ format_html function also calls marks_safe](https://docs.djangoproject.com/en/5.2/ref/utils/#django.utils.html.format_html). Marking a string as safe does not lead to security issues per se.
That happens only if we mark as safe input coming from the user, as it can be used for all sort of injections. But in this case we are generating the string without any user input, so it's safe. The security risk is imaginary here.